build.sbt

```diff
@@ -55,120 +82,168 @@ val settings = Seq(
+         | - Scala.js adds the "JS" suffix to a project name seen in build.sbt
+         | - Scala Native adds the "Native" suffix to a project name seen in build.sbt
+         | - Scala 2.12 adds the suffix "2_12" to a project name seen in build.sbt
+         | - Scala 2.13 adds no suffix to a project name seen in build.sbt
+         |
+         |When working with IntelliJ, edit "val ideScala = ..." and "val idePlatform = ..." within "val versions" in build.sbt to control which Scala version you're currently working with.""".stripMargin,
+    usefulTasks := Seq(
+      sbtwelcome.UsefulTask("listAll", "projects", "List all projects generated by the build matrix"),
+      sbtwelcome.UsefulTask("testAll", "test", "Compile and test all projects in all Scala versions and platforms (beware! it uses a lot of memory and might OOM!)"),
+      sbtwelcome.UsefulTask("stageRelease", "publishSigned", "Stage all versions for publishing"),
+      sbtwelcome.UsefulTask("publishRelease", "sonatypeBundleRelease", "Publish all artifacts staged for release"),
+      sbtwelcome.UsefulTask("ci-2_12", ciCommand("2_12"), "Checks formatting, run tests and compute core library coverage for Scala 2.12"),
+      sbtwelcome.UsefulTask("ci-2_13", ciCommand(""), "Checks formatting, run tests and compute core library coverage for Scala 2.13"),
+    )
   )
 
-lazy val chimney = crossProject(JSPlatform, JVMPlatform)
-  .crossType(CrossType.Pure)
-  .dependsOn(protos % "test->test")
+lazy val chimney = projectMatrix
+  .in(file("chimney"))
+  .someVariations(
+    versions.scalas,
+    versions.platforms
+  )(only1VersionInIDE: _*)
+  .disablePlugins(WelcomePlugin)
   .settings(
     moduleName := "chimney",
     name := "chimney",
-    description := "Scala library for boilerplate free data rewriting",
-    testFrameworks += new TestFramework("utest.runner.Framework"),
-    addCompilerPlugin("org.typelevel" % "kind-projector" % "0.13.2" cross CrossVersion.full)
+    description := "Scala library for boilerplate free data rewriting"
   )
   .settings(settings: _*)
   .settings(publishSettings: _*)
   .settings(dependencies: _*)
+  .dependsOn(protos % "test->test")
 
-lazy val chimneyJVM = chimney.jvm
-lazy val chimneyJS = chimney.js
-
-lazy val chimneyCats = crossProject(JSPlatform, JVMPlatform)
-  .crossType(CrossType.Pure)
-  .dependsOn(chimney % "test->test;compile->compile")
+lazy val chimneyCats = projectMatrix
+  .in(file("chimneyCats"))
+  .someVariations(
+    versions.scalas,
+    versions.platforms
+  )(only1VersionInIDE: _*)
+  .disablePlugins(WelcomePlugin)
   .settings(
     moduleName := "chimney-cats",
     name := "chimney-cats",
-    description := "Chimney module for validated transformers support",
-    testFrameworks += new TestFramework("utest.runner.Framework"),
-    addCompilerPlugin("org.typelevel" % "kind-projector" % "0.13.2" cross CrossVersion.full)
+    description := "Chimney module for validated transformers support"
```

Comment:
The purpose of the module is to provide better integration with cats library.

Reply:
I just left the old wording in this commit, but I agree that it's outdated so I updated it.